django_app/redbox_app/redbox_core/management/commands/delete_expired_data.py

```diff
@@ -0,0 +1,53 @@
+import logging
+from datetime import timedelta
+
+from botocore.exceptions import BotoCoreError
+from django.conf import settings
+from django.core.management import BaseCommand
+from django.utils import timezone
+from redbox_app.redbox_core.client import CoreApiClient
+from redbox_app.redbox_core.models import File, StatusEnum
+from requests.exceptions import RequestException
+
+logger = logging.getLogger(__name__)
+core_api = CoreApiClient(host=settings.CORE_API_HOST, port=settings.CORE_API_PORT)
+
+
+class Command(BaseCommand):
+    help = """This should be run daily per environment to remove expired data.
+    It removes Files that have exceeded their expiry date.
+    """
+
+    def handle(self, *_args, **_kwargs):
+        cutoff_date = timezone.now() - timedelta(seconds=settings.FILE_EXPIRY_IN_SECONDS)
+
+        self.stdout.write(self.style.NOTICE(f"Deleting Files expired before {cutoff_date}"))
+        counter = 0
+
+        for file in File.objects.filter(last_referenced__lt=cutoff_date):
+            logger.debug(
+                "Deleting file object %s, last_referenced %s",
+                file,
+                file.last_referenced,
+            )
+
+            try:
+                core_api.delete_file(file.core_file_uuid, file.user)
+                file.delete_from_s3()
```

Comment:
this could be part of the `File.delete` method 🤔 , probably doesnt add much and just over-complicates things?

Reply:
We currently have a File.delete that actually deletes the file, whereas this just removes the file contents but keeps a record of the File in Django (for auditing processes). I think this was a product decision a few weeks ago.